src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -84,8 +84,8 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
                     tokenFromOriginalTemplate = originalTemplateRoot.InsensitiveToken($"{originalResourcePath}.{remainingPathAtResourceScope}", InsensitivePathNotFoundBehavior.LastValid);
                 }
             }
-
-            return (tokenFromOriginalTemplate as IJsonLineInfo)?.LineNumber ?? 1;
+            
+            return (tokenFromOriginalTemplate as IJsonLineInfo)?.LineNumber + this.templateContext.Offset ?? 1; // Adds template's line number to an offset dependent on the parent (if applicable) template's position
```

Comment:
Can you add a test to JsonLineNumberResolverTests to validate the new behavior of the function?

Reply:
Done! 
Finally